### PR TITLE
Add reason for activity email

### LINKED ISSUE #5453: Show reason for activity email
For admins to distinguish whether they are actually involved in a review or getting an activity email because there are a member of the CC group, it would be useful to know why they got an activity email.

The email should not something like one of:
`You are receiving this mail because you are an author of the add-on.`
`You are receiving this mail because you reviewed this add-on.`
`You are receiving this mail because you are in the activity mail cc group.`

### PR DESCRIPTION
**Pull request by wagnerand**
Fix #5453.

### CODE REVIEW

**Comment by wagnerand** [comment]
Ehm, this still needs tests...

**Comment by wagnerand** [comment]
@eviljeff tests added. r?

File: src/olympia/activity/utils.py [hunk 1]
```diff
@@ -227,33 +227,47 @@ def log_and_notify(action, comments, note_creator, version, perm_setting=None,
         subject = u'Mozilla Add-ons: %s %s %s' % (
             version.addon.name, version.version, action.short)
     template = template_from_user(note_creator, version)
-    send_activity_mail(
-        subject, template.render(Context(author_context_dict)), version,
-        addon_authors, settings.EDITORS_EMAIL, perm_setting)
-    send_activity_mail(
-        subject, template.render(Context(reviewer_context_dict)), version,
-        reviewers, settings.EDITORS_EMAIL, perm_setting)
+
```
Comment (eviljeff) [inline src/olympia/activity/utils.py]: You don't need all this change - just do what I did already with `reviewer_context_dict` and create a new `_context_dict` for in staff_cc (you'll need to do some set logic to get a list of cc users who aren't in reviewers, addon_authors, task-user and the note_creator).  Then have a `third send_activity_mail`

Comment (wagnerand) [inline src/olympia/activity/utils.py]: 😕

**Comment by wagnerand** [comment]
@eviljeff Updated patch.

File: src/olympia/activity/utils.py [hunk 2]
```diff
@@ -205,34 +205,52 @@ def log_and_notify(action, comments, note_creator, version, perm_setting=None,
         task_user = set()
     # Collect reviewers on the thread (excl. the email sender and task user for
     # automated messages).
-    reviewers = ((log_users | staff_cc) - addon_authors - task_user -
-                 {note_creator})
+    reviewers = (log_users - addon_authors - task_user - {note_creator})
+    staff_cc = (staff - reviewers)
```
Comment (eviljeff) [inline src/olympia/activity/utils.py]: you need to `- addon_authors - {note_creator}` too or you'll get dupes for add-ons you're a developer on and your own replies.

File: src/olympia/activity/utils.py [hunk 3]
```diff
@@ -205,34 +205,52 @@ def log_and_notify(action, comments, note_creator, version, perm_setting=None,
         task_user = set()
     # Collect reviewers on the thread (excl. the email sender and task user for
     # automated messages).
-    reviewers = ((log_users | staff_cc) - addon_authors - task_user -
-                 {note_creator})
+    reviewers = (log_users - addon_authors - task_user - {note_creator})
```
Comment (eviljeff) [inline src/olympia/activity/utils.py]: you can drop the `(` `)` now

File: src/olympia/activity/utils.py (hunk 2 above)
Comment (wagnerand) [inline src/olympia/activity/utils.py]: Ah right, subtraction is hard!

**Pull request merged**